beancount.el

```diff
@@ -570,30 +578,40 @@ With an argument move to the previous non cleared transaction."
                     (complete-with-action action candidates string pred))))
             (list (match-beginning 1) (match-end 1) completion-table))))))))
 
-(defun beancount-collect (regexp n)
-  "Return an unique list of REGEXP group N in the current buffer."
+(defun beancount-collect-pos-alist (regexp n)
```

Comment:
Can you please move the refactoring to his own commit (and squash the other commits implementing the xref backend into a single commit)?

Reply:
This might be a bit too entangled by now. But let me try. 

Squashing is easy though :-)

Reply:
I squashed everything into one commit, and my git-fu is not good enough for extracting the refactoring bit into a separate commit. I should have done refactoring separately, sorry, I just went with the flow on this one.

Reply:
I split the commits and pushed to your branch.